### DISCUSSION TITLE
Authentication failed when install v3.2.6 maven package

### DISCUSSION THREAD

**Discussion by jjjjoooo** [argoproj/argo-workflows#7526] [Q&A] [answered]
I am trying to install java client library v3.2.6 maven package .
1. I added dependency to `pom.xml` as https://github.com/argoproj/argo-workflows/packages/1060616
2. `.m2/settings.xml` is same with https://github.com/argoproj/argo-workflows/blob/release-3.2/sdks/java/settings.xml.

But I have authentication issues, the error message is 
`Authentication failed for https://maven.pkg.github.com/argoproj/argo-workflows/io/argoproj/workflow/argo-client-java/v3.2.6/argo-client-java-v3.2.6.pom 401 Unauthorize`

It looks like the issue is caused by the password in .m2/settings.xml but I don't know how to fix it. May I have some suggestion for that?

Many thanks!

**Comment by terrytangyuan** [comment, marked as answer]
Not familiar with the Java SDK but I guess you'll need to update the `<servers>` field which has the wrong username `<username>alexec</username>`?

**Reply by jjjjoooo** [reply]
Hi @terrytangyuan,

Thank you for your reply! I just figured out a way to install the package,
1. Clone repo https://github.com/argoproj/argo-workflows to local.
2. Under folder `sdks/java`, run `mvn clean install`. Then there is a "argoproj" folder generated in `.m2` folder
3. Manually add `*.jar` files in https://github.com/argoproj/argo-workflows/packages/1060616 (depends on version) to `.m2/argoproj/workflow/argo-client-java/version` folder

**Reply by terrytangyuan** [reply]
Cool. Feel free to submit PRs to improve existing installation docs.

**Comment by alexec** [comment]
It is great to hear people want to use this! There is nothing special about our usage of Github repositories, so perhaps this will help?

https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-apache-maven-registry

**Reply by jjjjoooo** [reply]
I tried this but it doesn't work for me. It needs add package password in `~/.m2/settings.xml`. I don't know the password.

**Reply by simper66** [reply]
> There is nothing special about our usage of Github repositories

Just the opposite

**Comment by contacttapan** [comment]
Can we not publish the java client lib into Maven repo, Organization can't allow other repos easily to be mirrored. 

It would be nice to move this to Maven , specially this tool has good support for java ecosystem.

**Reply by agilgur5** [reply]
There's a feature request for that: https://github.com/argoproj/argo-workflows/issues/10895

> specially this tool has good support for java ecosystem.

The SDK is auto-generated from the OpenAPI spec, and it is honestly not the highest quality (people do report bugs with it that we can't necessarily fix because it's generated). Some users have reported better success in directly writing requests to the API than using the SDK.
A custom Java SDK written by the community might be a better route, similar to the [Hera Python SDK](https://github.com/argoproj-labs/hera/)
